### ISSUE TITLE
TypeError on SPARQLUpdateStore commit

### ISSUE THREAD

**Issue by bcroq** [RDFLib/rdflib#462]
`SPARQLUpdateStore` is not transaction aware so I wrap it in an `AuditableStore`. The problem is that `AuditableStore`'s commit calls `SPARQLUpdateStore`'s commit, which raises a `TypeError` exception:

```python

# ./fuseki-server --update --mem /auditable_sparql_store

import rdflib
from rdflib.plugins.stores.auditable import AuditableStore
from rdflib.plugins.stores.sparqlstore import SPARQLUpdateStore

store = SPARQLUpdateStore('http://fuseki.til.io:3030/auditable_sparql_store/query',
                          'http://fuseki.til.io:3030/auditable_sparql_store/update')
store = AuditableStore(store)

graph = rdflib.Graph(store, 'http://example.com/')

graph.add((rdflib.URIRef('http://example.com/s'), 
           rdflib.URIRef('http://example.com/p'), 
           rdflib.Literal('')))

graph.commit()
```

raises a `TypeError: The SPARQL Update store is not transaction aware!`

Maybe should we only get a warning instead of an exception?

**Comment by joernhees** [comment]
hmm, might make sense as the Store interface (https://github.com/RDFLib/rdflib/blob/master/rdflib/store.py#L364) doesn't just raise them if not implemented otherwise.

I'm not sure what the rational back then was to make SPARQLUpdateStore raise that exception while the default IOMemory store (which also isn't transaction aware) doesn't.

**Comment by joernhees** [comment]
this should've been fixed in 96551f9289f0cf15ecb88a83ef8de082649b1f46 in #486, which went into 4.2.1...
